cpu/em_cpufreq.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2017 IBM
+# Author: Shriya Kulkarni <shriyak@linux.vnet.ibm.com>
+import os
+import random
+import platform
+from avocado import Test
+from avocado import main
+from avocado.utils import process, distro, cpu
+from avocado.utils.software_manager import SoftwareManager
+
+
+class cpufreq(Test):
+    """
+    Test to validate the frequency transition.
+    """
+    def setup(self):
+        """
+        Verify the system is Baremetal and cpupower tool is installed.
+        """
+        if not os.path.exists('/sys/devices/system/cpu/cpu0/cpufreq'):
+            self.cancel('CPUFREQ is supported only on Power NV')
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        if 'Ubuntu' in detected_distro.name:
+            deps = ['linux-tools-common', 'linux-tools-%s'
+                    % platform.uname()[2]]
+        else:
+            deps = ['kernel-tools']
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def test(self):
+        """
+        Set the governor to userspace
+        choose random frequency and cpu
+        validate the frequency is set using ppc64_cpu tool.
+        """
+        self.cpu = 0
+        threshold = 10000
+        output = process.run("cpupower frequency-set -g userspace", shell=True)
+        cur_governor = self.cpu_freq_path('scaling_governor')
+        if (output.exit_status == 0) and ('userspace' == cur_governor):
```

Comment:
braces are not required here also. Better to place constant `'userspace'` in right hand side.